google-cloud-pubsub/samples/acceptance/helper.rb

```diff
@@ -18,6 +18,9 @@
 require "google/cloud/pubsub"
 require "securerandom"
 
+$tables = []
+$datasets = []
```

Comment:
Couldn't thing of a better way to clean up resource here.

Reply:
I suspect `after :all` doesn't do what you think it does. Minitest isn't rspec, and it ignores any argument passed to `before` or `after`. So `after :all` actually runs after each test, rather than once after all tests finish. Because your `cleanup_bq` doesn't clean up after itself (i.e. it doesn't empty the arrays after it deletes the items), it could end up attempting to delete items multiple times, and throw errors as a result.

So instead of collecting items to delete, I suggest having each test clean up its own mess. Maybe have `create_table` set instance variables for created tables and datasets, and then an `after` clause can just clean up anything set in those instance variables. And avoid the globals.

Reply:
Thanks Daniel.
Have updated accordingly.. even I was wary of using globals, using instance variables and cleanup after that particular test makes more sense. 
Also noticed with the `minitest-hooks` gem `:all` argument passed seems to be honoured.